x-pack/test/security_solution_endpoint/services/endpoint.ts

```diff
@@ -164,30 +198,35 @@ export class EndpointTestResources extends FtrService {
           },
         };
 
-    // If we have a specific number of endpoint hosts to check for, then use that number,
-    // else we just want to make sure the index has data, thus just having one in the index will do
-    const size = ids.length || 1;
-
-    await this.retry.waitFor('endpoint hosts', async () => {
-      try {
-        const searchResponse = await this.esClient.search({
-          index: metadataCurrentIndexPattern,
-          size,
-          body,
-          rest_total_hits_as_int: true,
-        });
+    await this.waitForIndex(ids, metadataCurrentIndexPattern, body, timeout);
```

Comment:
I'm actually thinking maybe this needs to also be the united index. The goal of htis method (`waitForEndpoints()`) is to wait until we are sure data exists to be shown on the Endpoint LIst (or returned on the `metadata` api), so I think (correct me if I'm wrong) the United target index is the one that supports that api, right?

Reply:
yeah, it definitely makes sense to wait for the united index. the issue I ran into here is that depending on the timing of the transform check point, it's possible that the united index doesn't get populated at all. I played around this in the test by only starting the united transform after verifying the `metadata_current` index was populated.
```
waitForEndpoints()
startTransform(UNITED_TRANSFORM)
waitForUnitedEndpoints()
```


I considered using a future timestamp for the metadata doc(s) + only checking for united index. while it works, it feels less reliable and similar to a timeout. what I think we can do in a follow up is update `loadEndpointData` to do:
```
stop united transform
generate + index docs
wait for metadata_current
start united transform
wait for metadata_united
```


I'm thinking we can also add an optional generateAndIndexDocs param for custom data so we can handle situations like the integration tests in this PR.